Assets/Internal/PluginDownloader.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class PluginDownloader
+{
+    private const string RegistryName = "Filta Artist Suite";
+    private const string RegistryUrl = "https://registry.npmjs.org";
+    private const string PluginUrl = "https://registry.npmjs.org/com.getfilta.artist-unityplug";
+    private const string RegistryScope = "com.getfilta.artist-unityplug";
+    
+    [InitializeOnLoadMethod]
+    private static async void UpdatePanel() {
+        ScopedRegistry filtaRegistry = new ScopedRegistry {
+            name = RegistryName,
+            url = RegistryUrl,
+            scopes = new[] {
+                RegistryScope
+            }
+        };
+        string manifestPath = Path.Combine(Application.dataPath, "..", "Packages/manifest.json");
+        string manifestJson = File.ReadAllText(manifestPath);
+        ManifestJson manifest = JsonConvert.DeserializeObject<ManifestJson>(manifestJson);
+        string version;
+        if (manifest.scopedRegistries.FindIndex((registry => registry.url == RegistryUrl)) == -1) {
+            version = await GetVersion();
+            if (String.IsNullOrEmpty(version)) {
+                Retry();
+                return;
+            }
+            manifest.scopedRegistries.Add(filtaRegistry);
+        } else {
+            return;
```

Comment:
i feel like im confusing something - should this actually be a return?

Reply:
yes, it should be a return. this UpdatePanel() Function is called OnLoad, which is any time the user boots up Unity. 

line 32 checks if the plugin has already been added to manifest. if it hasn't, do all the adding and magic.

but if the plugin is already added to manifest, do nothing. that's why the return is there in line 40.

Reply:
Ah - my confusion was "Wouldn't you want to check the version as well to see if it needs to be updated?"

but the answer to that is actually "no"